pkg/ipam/types.go

```diff
@@ -118,19 +116,6 @@ type IPAM struct {
 	nodeDiscovery  Owner
 }
 
-// DebugStatus implements debug.StatusObject to provide debug status collection
-// ability
-func (ipam *IPAM) DebugStatus() string {
```

Comment:
I actually wasn't aware of this functionality. While I don't object to removing previously broken code, I think it would be very useful to dump the internal state of the allocators at least. But doesn't need to be part of this PR, we can track that in a follow-up item.

Reply:
Me neither  :smile: 

The problem of using `go-spew` and just passing the whole object is that we easily end up with memory issues (that's why i removed it in the first place). And the output easily gets overwhelming and is no longer useful.

IMO it's up to each "component" to register itself and only add the relevant output to the debug status. Hence `IPAM` or the individual allocators would register themselves individually.

A proposal how we could "fix" this in the current PR just for the IPAM bits - by manually selecting the relevant data (without passing the dependencies).

```
func (ipam *IPAM) DebugStatus() string {
	ipam.allocatorMutex.RLock()
	str := spew.Sdump(
		"owners", ipam.owner,
		"expiration timers", ipam.expirationTimers,
		"excluded ips", ipam.excludedIPs,
	)
	ipam.allocatorMutex.RUnlock()
	return str
}
```

->

Example output
```
cilium-dbg debuginfo
...
#### ipam

(string) (len=6) "owners"
(map[ipam.Pool]map[string]string) (len=1) {
 (ipam.Pool) (len=7) default: (map[string]string) (len=7) {
  (string) (len=11) "10.244.1.84": (string) (len=42) "cilium-test/echo-same-node-9975f89cd-mvglg",
  (string) (len=12) "10.244.1.183": (string) (len=6) "router",
  (string) (len=12) "10.244.1.113": (string) (len=7) "ingress",
  (string) (len=12) "10.244.1.160": (string) (len=6) "health",
  (string) (len=11) "10.244.1.82": (string) (len=41) "kube-system/hubble-relay-5f75dc55b6-25xfx",
  (string) (len=11) "10.244.1.20": (string) (len=35) "cilium-test/client-856666cb56-7x8tf",
  (string) (len=12) "10.244.1.240": (string) (len=35) "cilium-test/client2-b6f5f8c45-p7r8c"
 }
}
(string) (len=17) "expiration timers"
(map[ipam.timerKey]ipam.expirationTimer) {
}
(string) (len=12) "excluded ips"
(map[string]string) (len=1) {
 (string) (len=18) "default:172.19.0.3": (string) (len=7) "node-ip"
}
...
```

Follow-up items could add this for other "inner" components where useful. WDYT?

Reply:
> Follow-up items could add this for other "inner" components where useful. WDYT?

I think that makes sense, thanks for experimenting!

Reply:
refactored as described above.